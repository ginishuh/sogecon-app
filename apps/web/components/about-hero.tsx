import Image from 'next/image';
import React from 'react';

type AboutHeroProps = {
  title: string;
  description: string;
  image: {
    src: string;
    alt: string;
    priority?: boolean;
  };
};

export function AboutHero({ title, description, image }: AboutHeroProps) {
  return (
    <section aria-labelledby="about-hero-heading" className="about-hero">
      <div className="about-hero__text">
        <p className="about-hero__eyebrow">총동문회 소개</p>
        <h1 id="about-hero-heading" className="about-hero__title">
          {title}
        </h1>
        <p className="about-hero__description">{description}</p>
      </div>
      <div className="about-hero__image">
        <Image
          alt={image.alt}
          src={image.src}
          width={640}
          height={480}
          className="h-full w-full rounded-xl object-cover shadow-md"
          priority={image.priority}
        />
      </div>
    </section>
  );
}
