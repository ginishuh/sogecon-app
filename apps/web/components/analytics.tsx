import Script from 'next/script';
import React from 'react';

const measurementId = process.env.NEXT_PUBLIC_ANALYTICS_ID;

export function Analytics() {
  if (!measurementId) {
    return null;
  }

  const bootstrap = `
    window.dataLayer = window.dataLayer || [];
    function gtag(){window.dataLayer.push(arguments);}
    gtag('js', new Date());
    gtag('config', '${measurementId}', { anonymize_ip: true });
  `;

  return (
    <>
      {/* 프로덕션 도메인 확정 전까지는 환경변수로 가드하고, 기본 페이지뷰만 기록 */}
      <Script id="ga-loader" src={`https://www.googletagmanager.com/gtag/js?id=${measurementId}`} strategy="afterInteractive" />
      <Script id="ga-inline" strategy="afterInteractive">{bootstrap}</Script>
    </>
  );
}
