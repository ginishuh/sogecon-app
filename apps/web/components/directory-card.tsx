import React, { useState } from 'react';
import type { Member } from '../services/members';

type Props = {
  member: Member;
};

export default function DirectoryCard({ member }: Props) {
  const [open, setOpen] = useState(false);
  const panelId = `dir-card-${member.id}`;
  return (
    <article className="rounded-2xl border border-neutral-border bg-white p-4 shadow-sm">
      <header className="flex items-start justify-between gap-3">
        <div className="min-w-0">
          <h3 className="text-base font-semibold text-neutral-ink">
            {member.name}
            <span className="ml-2 align-middle text-xs font-medium text-neutral-muted">{member.cohort}기</span>
          </h3>
          <p className="truncate text-xs text-neutral-muted">
            {member.email}
          </p>
          <p className="mt-1 text-sm text-neutral-ink">
            {member.company ?? '-'}
          </p>
        </div>
        <button
          type="button"
          aria-expanded={open}
          aria-controls={panelId}
          onClick={() => setOpen((v) => !v)}
          className="rounded border border-neutral-border px-3 py-1.5 text-sm text-neutral-ink hover:bg-neutral-surface"
        >
          {open ? '접기' : '자세히'}
        </button>
      </header>
      <section id={panelId} role="region" aria-label="상세 정보" hidden={!open} className="mt-3 text-sm text-neutral-ink">
        <dl className="grid grid-cols-2 gap-2">
          <div>
            <dt className="text-xs text-neutral-muted">전공</dt>
            <dd>{member.major ?? '-'}</dd>
          </div>
          <div>
            <dt className="text-xs text-neutral-muted">업종</dt>
            <dd>{member.industry ?? '-'}</dd>
          </div>
          <div>
            <dt className="text-xs text-neutral-muted">공개 범위</dt>
            <dd className="text-xs uppercase text-neutral-muted">{member.visibility}</dd>
          </div>
        </dl>
      </section>
    </article>
  );
}

