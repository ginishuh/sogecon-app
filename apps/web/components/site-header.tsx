"use client";

import React, { useState } from 'react';
import Link from 'next/link';
import type { Route } from 'next';

import { HeaderAuth } from './header-auth';
import { NotifyCTA } from './notify-cta';
import { RequireMember } from './require-member';
import { RequireAdmin } from './require-admin';

type LinkItem = {
  href: Route;
  label: string;
};

const PRIMARY_LINKS: LinkItem[] = [
  { href: '/', label: '홈' },
  { href: '/posts', label: '총원우회 소식' },
  { href: '/events', label: '행사' }
];

const ABOUT_LINKS: LinkItem[] = [
  { href: '/about/greeting', label: '회장 인사말' },
  { href: '/about/org', label: '조직도' },
  { href: '/about/history', label: '역대 회장단' }
];

const SUPPORT_LINKS: LinkItem[] = [
  { href: '/faq', label: 'FAQ' },
  { href: '/privacy', label: '개인정보 처리방침' },
  { href: '/terms', label: '이용약관' }
];

const ADMIN_LINKS: LinkItem[] = [
  { href: '/admin/notifications', label: '알림(Admin)' },
  { href: '/posts/new', label: '게시글 작성' },
  { href: '/events/new', label: '행사 생성' }
];

export function SiteHeader() {
  const [open, setOpen] = useState(false);

  const closeMenu = () => setOpen(false);

  return (
    <header className="border-b border-neutral-border bg-white shadow-sm">
      <div className="mx-auto flex max-w-6xl items-center justify-between gap-4 px-4 py-4 md:px-6">
        <Link href="/" className="flex items-center gap-2 text-lg font-semibold text-brand-primary">
          <span
            aria-label="총원우회 홈으로 이동"
            className="rounded-full bg-brand-primary px-2 py-1 text-sm text-white"
          >
            SG
          </span>
          <span className="tracking-tight text-neutral-ink">서강대 경제대학원 총원우회</span>
        </Link>
        <button
          type="button"
          className="inline-flex items-center justify-center rounded-md border border-neutral-border px-3 py-2 text-sm text-neutral-ink transition hover:bg-brand-surface focus:outline-none focus-visible:ring-2 focus-visible:ring-brand-accent md:hidden"
          aria-controls="primary-navigation"
          aria-expanded={open}
          aria-label={open ? '전체 메뉴 닫기' : '전체 메뉴 열기'}
          onClick={() => setOpen((prev) => !prev)}
        >
          <span className="sr-only">메뉴 토글</span>
          <svg aria-hidden="true" className="h-5 w-5" fill="none" stroke="currentColor" strokeWidth={2} viewBox="0 0 24 24">
            {open ? (
              <path strokeLinecap="round" strokeLinejoin="round" d="M6 18L18 6M6 6l12 12" />
            ) : (
              <path strokeLinecap="round" strokeLinejoin="round" d="M4 6h16M4 12h16M4 18h16" />
            )}
          </svg>
        </button>
        <div className="hidden flex-1 items-center justify-between md:flex">
          <nav
            aria-label="주 메뉴"
            className="flex flex-1 items-center justify-between gap-10 text-sm text-neutral-muted"
          >
            <ul className="flex items-center gap-5">
              {PRIMARY_LINKS.map((link) => (
                <li key={link.href}>
                  <Link className="transition hover:text-brand-primary" href={link.href}>
                    {link.label}
                  </Link>
                </li>
              ))}
            </ul>
            <div className="flex items-start gap-8 text-xs">
              <div aria-label="총원우회 소개" className="flex flex-col gap-2">
                <span className="text-[11px] font-semibold uppercase tracking-wide text-neutral-muted">
                  총원우회 소개
                </span>
                <ul className="flex flex-col gap-2">
                  {ABOUT_LINKS.map((link) => (
                    <li key={link.href}>
                      <Link className="transition hover:text-brand-primary" href={link.href}>
                        {link.label}
                      </Link>
                    </li>
                  ))}
                </ul>
              </div>
              <div aria-label="고객 지원" className="flex flex-col gap-2">
                <span className="text-[11px] font-semibold uppercase tracking-wide text-neutral-muted">
                  고객 지원
                </span>
                <ul className="flex flex-col gap-2">
                  {SUPPORT_LINKS.map((link) => (
                    <li key={link.href}>
                      <Link className="transition hover:text-brand-primary" href={link.href}>
                        {link.label}
                      </Link>
                    </li>
                  ))}
                </ul>
              </div>
              <RequireAdmin>
                <div aria-label="관리자 메뉴" className="flex flex-col gap-2">
                  <span className="text-[11px] font-semibold uppercase tracking-wide text-neutral-muted">
                    관리자
                  </span>
                  <ul className="flex flex-col gap-2">
                    {ADMIN_LINKS.map((link) => (
                      <li key={link.href}>
                        <Link className="transition hover:text-brand-primary" href={link.href}>
                          {link.label}
                        </Link>
                      </li>
                    ))}
                  </ul>
                </div>
              </RequireAdmin>
            </div>
          </nav>
          <div className="flex items-center gap-4">
            <RequireMember>
              <NotifyCTA />
            </RequireMember>
            <HeaderAuth />
          </div>
        </div>
      </div>
      <div
        id="primary-navigation"
        className={`md:hidden ${open ? 'block' : 'hidden'} border-t border-neutral-border bg-white px-4 pb-4`}
      >
        <nav aria-label="모바일 주 메뉴" className="flex flex-col gap-4 pt-4 text-sm text-neutral-muted">
          <ul className="grid gap-3" aria-label="주요 링크">
            {PRIMARY_LINKS.map((link) => (
              <li key={link.href}>
                <Link
                  className="block rounded-md border border-neutral-border px-3 py-2 hover:border-brand-primary hover:text-brand-primary"
                  href={link.href}
                  onClick={closeMenu}
                >
                  {link.label}
                </Link>
              </li>
            ))}
          </ul>
          <section aria-label="총원우회 소개 링크" className="space-y-2">
            <h2 className="text-xs font-semibold uppercase text-neutral-muted">총원우회 소개</h2>
            <ul className="grid gap-2">
              {ABOUT_LINKS.map((link) => (
                <li key={link.href}>
                  <Link
                    className="block rounded-md px-3 py-2 hover:bg-brand-surface hover:text-brand-primary"
                    href={link.href}
                    onClick={closeMenu}
                  >
                    {link.label}
                  </Link>
                </li>
              ))}
            </ul>
          </section>
          <section aria-label="고객 지원 링크" className="space-y-2">
            <h2 className="text-xs font-semibold uppercase text-neutral-muted">고객 지원</h2>
            <ul className="grid gap-2">
              {SUPPORT_LINKS.map((link) => (
                <li key={link.href}>
                  <Link
                    className="block rounded-md px-3 py-2 hover:bg-brand-surface hover:text-brand-primary"
                    href={link.href}
                    onClick={closeMenu}
                  >
                    {link.label}
                  </Link>
                </li>
              ))}
            </ul>
          </section>
          <RequireAdmin
            fallback={
              <div className="text-xs text-neutral-muted">
                관리자 전용 메뉴는 로그인 후 확인할 수 있습니다.
              </div>
            }
          >
            <section aria-label="관리자 링크" className="space-y-2">
              <h2 className="text-xs font-semibold uppercase text-neutral-muted">관리자</h2>
              <ul className="grid gap-2">
                {ADMIN_LINKS.map((link) => (
                  <li key={link.href}>
                    <Link
                      className="block rounded-md px-3 py-2 hover:bg-brand-surface hover:text-brand-primary"
                      href={link.href}
                      onClick={closeMenu}
                    >
                      {link.label}
                    </Link>
                  </li>
                ))}
              </ul>
            </section>
          </RequireAdmin>
          <div className="flex flex-col gap-2 border-t border-neutral-border pt-4">
            <RequireMember>
              <NotifyCTA />
            </RequireMember>
            <HeaderAuth />
          </div>
        </nav>
      </div>
    </header>
  );
}
