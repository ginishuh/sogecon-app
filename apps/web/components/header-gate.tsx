"use client";

import { SiteHeader } from './site-header';

// 로그인 페이지에서도 헤더를 그대로 노출합니다.
export function HeaderGate() {
  return <SiteHeader />;
}
