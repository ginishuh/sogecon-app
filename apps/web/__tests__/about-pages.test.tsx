import { cleanup, fireEvent, render, screen, within } from '@testing-library/react';
import React, { type ReactNode } from 'react';
import { afterEach, describe, expect, it, vi } from 'vitest';

import GreetingPage from '../app/about/greeting/page';
import OrgPage from '../app/about/org/page';
import HistoryPage from '../app/about/history/page';
import { SiteHeader } from '../components/site-header';
import { QueryClient, QueryClientProvider } from '@tanstack/react-query';

vi.mock('../components/header-auth', () => ({
  HeaderAuth: () => <div data-testid="header-auth">header-auth-placeholder</div>
}));

vi.mock('../components/notify-cta', () => ({
  NotifyCTA: () => <button type="button">알림 설정</button>
}));

vi.mock('../components/require-member', () => ({
  RequireMember: ({ children, fallback }: { children: ReactNode; fallback?: ReactNode }) => (
    <>{children || fallback || null}</>
  )
}));

vi.mock('../components/require-admin', () => ({
  RequireAdmin: ({ children, fallback }: { children: ReactNode; fallback?: ReactNode }) => (
    <>{children || fallback || null}</>
  )
}));

afterEach(() => {
  cleanup();
});

describe('About static pages', () => {
  it('renders greeting page hero and sections', () => {
    const { asFragment } = render(<GreetingPage />);
    expect(screen.getByRole('heading', { level: 1 })).toHaveTextContent('회장 인사말');
    expect(screen.getByLabelText('총동문회 비전 다이어그램')).toBeInTheDocument();
    expect(screen.getByRole('heading', { name: '디지털 허브 완성' })).toBeInTheDocument();
    expect(screen.getByRole('list', { name: '총동문회 운영 원칙 목록' })).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders organization page with contact list', () => {
    const { asFragment } = render(<OrgPage />);
    expect(screen.getByRole('heading', { name: '분과별 주요 역할' })).toBeInTheDocument();
    expect(screen.getByLabelText('사무국 연락처 정보')).toBeInTheDocument();
    expect(screen.getByText('소통분과')).toBeInTheDocument();
    expect(screen.getByText('대표 전화')).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders history page timeline and archive', () => {
    const { asFragment } = render(<HistoryPage />);
    expect(screen.getByRole('list', { name: '총동문회 주요 연혁' })).toBeInTheDocument();
    expect(screen.getByRole('heading', { name: '공식 웹 런치' })).toBeInTheDocument();
    expect(screen.getByText('16대 회장 김서강')).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });
});

describe('SiteHeader navigation', () => {
  it('exposes about links through accessible navigation', () => {
    const qc = new QueryClient();
    render(
      <QueryClientProvider client={qc}>
        <SiteHeader />
      </QueryClientProvider>
    );
    const toggleButton = screen.getByLabelText('전체 메뉴 열기');
    fireEvent.click(toggleButton);

    expect(toggleButton).toHaveAttribute('aria-expanded', 'true');
    const mobileNav = screen.getByLabelText('모바일 주 메뉴');
    expect(within(mobileNav).getByRole('link', { name: '회장 인사말' })).toHaveAttribute('href', '/about/greeting');
    expect(within(mobileNav).getByRole('link', { name: '조직도' })).toHaveAttribute('href', '/about/org');
    expect(within(mobileNav).getByRole('link', { name: '역대 회장단' })).toHaveAttribute('href', '/about/history');
    expect(within(mobileNav).getByRole('link', { name: 'FAQ' })).toHaveAttribute('href', '/faq');
    expect(within(mobileNav).getByRole('link', { name: '개인정보 처리방침' })).toHaveAttribute('href', '/privacy');
    expect(within(mobileNav).getByRole('link', { name: '이용약관' })).toHaveAttribute('href', '/terms');
  });

  it('marks navigation landmarks for desktop layout', () => {
    const qc = new QueryClient();
    render(
      <QueryClientProvider client={qc}>
        <SiteHeader />
      </QueryClientProvider>
    );
    // 데스크톱 주 메뉴 랜드마크와 기본 링크 확인
    expect(screen.getByLabelText('주 메뉴')).toBeInTheDocument();
    const toggleButton = screen.getByLabelText('전체 메뉴 열기');
    expect(toggleButton).toHaveAttribute('aria-controls', 'primary-navigation');
    const mainNav = screen.getByLabelText('주 메뉴');
    expect(within(mainNav).getByRole('link', { name: '홈' })).toHaveAttribute('href', '/');
    expect(within(mainNav).getByRole('link', { name: '총동문회 소식' })).toHaveAttribute('href', '/posts');
    expect(within(mainNav).getByRole('link', { name: '행사' })).toHaveAttribute('href', '/events');
  });
});
