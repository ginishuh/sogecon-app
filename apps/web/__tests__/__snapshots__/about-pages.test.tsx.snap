// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`About static pages > renders greeting page hero and sections 1`] = `
<DocumentFragment>
  <div
    class="about-page"
  >
    <section
      aria-labelledby="about-hero-heading"
      class="about-hero"
    >
      <div
        class="about-hero__text"
      >
        <p
          class="about-hero__eyebrow"
        >
          총동문회 소개
        </p>
        <h1
          class="about-hero__title"
          id="about-hero-heading"
        >
          회장 인사말
        </h1>
        <p
          class="about-hero__description"
        >
          총동문회 16대 회장 김서강입니다. 2025년 웹 서비스 런치와 함께, 동문 누구나 쉽게 소식을 접하고 의견을 나눌 수 있는 열린 커뮤니티를 만들겠습니다.
        </p>
      </div>
      <div
        class="about-hero__image"
      >
        <img
          alt="총동문회 서강대 캠퍼스 풍경 삽화"
          class="h-full w-full rounded-xl object-cover shadow-md"
          height="480"
          src="/images/about/greeting-hero.svg"
          width="640"
        />
      </div>
    </section>
    <section
      aria-labelledby="greeting-vision-title"
      class="about-section"
    >
      <div
        class="about-section__content"
      >
        <h2
          class="about-section__heading"
          id="greeting-vision-title"
        >
          동문이 함께 만드는 미래
        </h2>
        <p>
          지난 30년 동안 총동문회는 서로의 지식과 경험을 나누며 성장해 왔습니다. 이제는 모바일과 웹 어디서나 같은 정보를 볼 수 있도록 디지털 전환을 완성해, 멀리 떨어진 동문도 실시간으로 소식을 접할 수 있는 환경을 구축하고 있습니다.
        </p>
        <p>
          2025년에는 정기총회와 산업별 라운드테이블, ESG 프로젝트를 아우르는 ‘공유 달력’을 운영해 참여 동선을 단순화할 예정입니다. 동문 여러분 한 분, 한 분의 제안이 정책과 프로그램으로 이어지도록 의견 수렴 창구도 상시로 열어 두겠습니다.
        </p>
      </div>
      <figure
        aria-label="총동문회 비전 다이어그램"
        class="about-section__figure"
      >
        <img
          alt="총동문회의 비전을 상징하는 원형 다이어그램"
          class="h-auto w-full rounded-lg object-cover shadow-sm"
          height="360"
          src="/images/about/greeting-detail.svg"
          width="520"
        />
      </figure>
    </section>
    <section
      aria-labelledby="greeting-commitments-title"
      class="about-section about-section--stacked"
    >
      <div
        class="about-section__content"
      >
        <h2
          class="about-section__heading"
          id="greeting-commitments-title"
        >
          약속드리는 세 가지 방향
        </h2>
        <p>
          총동문회가 추구하는 핵심 가치를 세 가지로 정리했습니다. 모든 사업과 모임은 이 가치 위에서 기획하고, 결과를 투명하게 공유하겠습니다.
        </p>
      </div>
      <ul
        aria-label="총동문회 운영 원칙 목록"
        class="about-section__list"
      >
        <li
          class="about-section__card"
        >
          <h3
            class="about-section__card-title"
          >
            디지털 허브 완성
          </h3>
          <p>
            2025년 하반기 웹 런치를 통해 언제 어디서든 행사·공지·수첩을 확인할 수 있는 원스톱 허브를 완성하겠습니다.
          </p>
        </li>
        <li
          class="about-section__card"
        >
          <h3
            class="about-section__card-title"
          >
            투명한 재정과 보고
          </h3>
          <p>
            회비·기금 집행 내역을 분기별 리포트와 데이터 대시보드로 공개하고, 회원 질문에 48시간 이내 응답하는 체계를 유지하겠습니다.
          </p>
        </li>
        <li
          class="about-section__card"
        >
          <h3
            class="about-section__card-title"
          >
            함께 성장하는 커뮤니티
          </h3>
          <p>
            현직·동문 멘토링과 산업별 라운드테이블을 정례화해 기수와 세대를 아우르는 실질적인 성장을 지원하겠습니다.
          </p>
        </li>
      </ul>
    </section>
  </div>
</DocumentFragment>
`;

exports[`About static pages > renders history page timeline and archive 1`] = `
<DocumentFragment>
  <div
    class="about-page"
  >
    <section
      aria-labelledby="about-hero-heading"
      class="about-hero"
    >
      <div
        class="about-hero__text"
      >
        <p
          class="about-hero__eyebrow"
        >
          총동문회 소개
        </p>
        <h1
          class="about-hero__title"
          id="about-hero-heading"
        >
          역대 회장단과 연혁
        </h1>
        <p
          class="about-hero__description"
        >
          총동문회는 1994년 창립 이후 30년 넘게 동문 교류와 사회 공헌을 이어왔습니다. 연혁과 역대 회장단의 주요 활동을 한눈에 정리했습니다.
        </p>
      </div>
      <div
        class="about-hero__image"
      >
        <img
          alt="총동문회 연혁을 상징하는 타임라인 삽화"
          class="h-full w-full rounded-xl object-cover shadow-md"
          height="480"
          src="/images/about/history-hero.svg"
          width="640"
        />
      </div>
    </section>
    <section
      aria-labelledby="history-timeline"
      class="about-section about-section--timeline"
    >
      <div
        class="about-section__content"
      >
        <h2
          class="about-section__heading"
          id="history-timeline"
        >
          주요 연혁
        </h2>
        <p>
          동문회의 발자취를 간략히 정리했습니다. 연간 사업 보고서와 자료실에서 더 자세한 기록을 확인할 수 있으며, 2025년부터는 주요 지표를 데이터 보드로 시각화해 제공할 예정입니다.
        </p>
      </div>
      <ol
        aria-label="총동문회 주요 연혁"
        class="about-timeline"
      >
        <li
          class="about-timeline__item"
        >
          <div
            aria-hidden="true"
            class="about-timeline__marker"
          />
          <div
            class="about-timeline__content"
          >
            <span
              class="about-timeline__year"
            >
              1994
            </span>
            <h3
              class="about-timeline__title"
            >
              총동문회 창립
            </h3>
            <p>
              첫 4개 기수가 연합해 총동문회를 창립하고, 서강대학교 경제대학원 동문 네트워크의 기반을 마련했습니다.
            </p>
          </div>
        </li>
        <li
          class="about-timeline__item"
        >
          <div
            aria-hidden="true"
            class="about-timeline__marker"
          />
          <div
            class="about-timeline__content"
          >
            <span
              class="about-timeline__year"
            >
              2005
            </span>
            <h3
              class="about-timeline__title"
            >
              장학기금 조성
            </h3>
            <p>
              동문 기부로 장학기금을 조성해 재학생 후배 지원을 시작했습니다. 현재는 연간 12명에게 장학금을 지급합니다.
            </p>
          </div>
        </li>
        <li
          class="about-timeline__item"
        >
          <div
            aria-hidden="true"
            class="about-timeline__marker"
          />
          <div
            class="about-timeline__content"
          >
            <span
              class="about-timeline__year"
            >
              2016
            </span>
            <h3
              class="about-timeline__title"
            >
              디지털 커뮤니티 전환
            </h3>
            <p>
              회원 포털과 모바일 알림 시스템을 도입해 행사 신청, 회비 납부, 커뮤니티 운영을 온라인으로 전환했습니다.
            </p>
          </div>
        </li>
        <li
          class="about-timeline__item"
        >
          <div
            aria-hidden="true"
            class="about-timeline__marker"
          />
          <div
            class="about-timeline__content"
          >
            <span
              class="about-timeline__year"
            >
              2022
            </span>
            <h3
              class="about-timeline__title"
            >
              데이터 거버넌스 강화
            </h3>
            <p>
              회원 데이터 거버넌스 정책과 접근 로그 감사를 도입해 개인정보 보호와 투명한 데이터 사용 체계를 구축했습니다.
            </p>
          </div>
        </li>
        <li
          class="about-timeline__item"
        >
          <div
            aria-hidden="true"
            class="about-timeline__marker"
          />
          <div
            class="about-timeline__content"
          >
            <span
              class="about-timeline__year"
            >
              2025
            </span>
            <h3
              class="about-timeline__title"
            >
              공식 웹 런치
            </h3>
            <p>
              웹·모바일 통합 경험을 제공하는 공식 웹을 오픈하고, 수첩/공지/행사/소개를 한 번에 연결하도록 개편했습니다.
            </p>
          </div>
        </li>
      </ol>
    </section>
    <section
      aria-labelledby="history-presidents"
      class="about-section about-section--stacked"
    >
      <div
        class="about-section__content"
      >
        <h2
          class="about-section__heading"
          id="history-presidents"
        >
          역대 회장단
        </h2>
        <p>
          지난 6년간 총동문회를 이끌어 주신 회장단입니다. 각 회장단의 주요 사업과 과제를 간략히 소개합니다.
        </p>
      </div>
      <ul
        aria-label="역대 회장단 소개"
        class="about-section__list"
      >
        <li
          class="about-section__card"
        >
          <h3
            class="about-section__card-title"
          >
            13대 회장 박지원
          </h3>
          <dl
            class="about-section__dl"
          >
            <div>
              <dt>
                활동 기간
              </dt>
              <dd>
                2018 - 2019
              </dd>
            </div>
            <div>
              <dt>
                주요 과제
              </dt>
              <dd>
                디지털 전환, 회원 데이터베이스 정비
              </dd>
            </div>
          </dl>
        </li>
        <li
          class="about-section__card"
        >
          <h3
            class="about-section__card-title"
          >
            14대 회장 이상훈
          </h3>
          <dl
            class="about-section__dl"
          >
            <div>
              <dt>
                활동 기간
              </dt>
              <dd>
                2020 - 2021
              </dd>
            </div>
            <div>
              <dt>
                주요 과제
              </dt>
              <dd>
                비대면 운영 체계 확립, 온라인 세미나 확대
              </dd>
            </div>
          </dl>
        </li>
        <li
          class="about-section__card"
        >
          <h3
            class="about-section__card-title"
          >
            15대 회장 정아름
          </h3>
          <dl
            class="about-section__dl"
          >
            <div>
              <dt>
                활동 기간
              </dt>
              <dd>
                2022 - 2023
              </dd>
            </div>
            <div>
              <dt>
                주요 과제
              </dt>
              <dd>
                장학 사업 확장, ESG 프로젝트 준비
              </dd>
            </div>
          </dl>
        </li>
        <li
          class="about-section__card"
        >
          <h3
            class="about-section__card-title"
          >
            16대 회장 김서강
          </h3>
          <dl
            class="about-section__dl"
          >
            <div>
              <dt>
                활동 기간
              </dt>
              <dd>
                2024 - 2025
              </dd>
            </div>
            <div>
              <dt>
                주요 과제
              </dt>
              <dd>
                웹 런치, 데이터 투명성, 산업별 라운드테이블 상시화
              </dd>
            </div>
          </dl>
        </li>
      </ul>
    </section>
    <section
      aria-labelledby="history-archive"
      class="about-section"
    >
      <div
        class="about-section__content"
      >
        <h2
          class="about-section__heading"
          id="history-archive"
        >
          자료실 미리보기
        </h2>
        <p>
          연간 사업 보고서, 회계 감사 보고, 총회 의사록 등은 로그인한 회원에게 순차적으로 열람 권한을 제공할 예정입니다. 정식 오픈 전까지는 핵심 항목만 요약으로 제공하며, 열람 신청이 많은 문서는 우선 공개합니다.
        </p>
        <p>
          자료 공개 범위에 대한 의견은 언제든 사무국으로 전해주세요. 공개 시기와 형식을 구성원과 함께 결정하겠습니다.
        </p>
      </div>
      <figure
        aria-label="자료 보관소 이미지"
        class="about-section__figure"
      >
        <img
          alt="자료실 서가를 묘사한 일러스트"
          class="h-auto w-full rounded-lg object-cover shadow-sm"
          height="360"
          src="/images/about/history-archive.svg"
          width="520"
        />
      </figure>
    </section>
  </div>
</DocumentFragment>
`;

exports[`About static pages > renders organization page with contact list 1`] = `
<DocumentFragment>
  <div
    class="about-page"
  >
    <section
      aria-labelledby="about-hero-heading"
      class="about-hero"
    >
      <div
        class="about-hero__text"
      >
        <p
          class="about-hero__eyebrow"
        >
          총동문회 소개
        </p>
        <h1
          class="about-hero__title"
          id="about-hero-heading"
        >
          조직도
        </h1>
        <p
          class="about-hero__description"
        >
          총동문회는 회장단, 집행부, 감사단, 자문위원으로 구성되어 각자의 책임을 분담합니다. 모든 회의록과 결산 자료는 회원 전용 아카이브에 공개합니다.
        </p>
      </div>
      <div
        class="about-hero__image"
      >
        <img
          alt="총동문회 조직 구성을 보여주는 삽화"
          class="h-full w-full rounded-xl object-cover shadow-md"
          height="480"
          src="/images/about/org-hero.svg"
          width="640"
        />
      </div>
    </section>
    <section
      aria-labelledby="org-structure"
      class="about-section"
    >
      <div
        class="about-section__content"
      >
        <h2
          class="about-section__heading"
          id="org-structure"
        >
          회장단과 집행부
        </h2>
        <p>
          회장단은 회장, 수석부회장, 부회장단, 감사단으로 구성되어 전략 방향과 연간 사업계획을 수립합니다. 집행부는 실행 조직으로서 각 분과의 목표를 조정하고, 회원 접점을 통해 수집된 피드백을 정리해 회장단에 보고합니다.
        </p>
        <p>
          모든 분과는 담당자를 두 명 이상 배치해 견제와 협업 구조를 유지하며, 주요 안건은 월 1회 열리는 운영위원회에서 논의 후 공지합니다. 분기별로 공개하는 리포트에는 KPI 달성 현황과 다음 분기 계획이 포함됩니다.
        </p>
      </div>
      <figure
        aria-label="총동문회 조직 계층 구조"
        class="about-section__figure"
      >
        <img
          alt="회장단, 집행부, 자문위원으로 구성된 조직도"
          class="h-auto w-full rounded-lg object-cover shadow-sm"
          height="360"
          src="/images/about/org-chart.svg"
          width="520"
        />
      </figure>
    </section>
    <section
      aria-labelledby="org-committees"
      class="about-section about-section--stacked"
    >
      <div
        class="about-section__content"
      >
        <h2
          class="about-section__heading"
          id="org-committees"
        >
          분과별 주요 역할
        </h2>
        <p>
          각 분과는 연초에 정량 목표와 예산 사용 계획을 공지하고, 분기별 성과와 지출 내역을 대시보드와 PDF 리포트로 공유합니다. 주요 지표는 KPI 보드에서 실시간으로 확인할 수 있습니다.
        </p>
      </div>
      <ul
        aria-label="분과 역할 목록"
        class="about-section__list"
      >
        <li
          class="about-section__card"
        >
          <h3
            class="about-section__card-title"
          >
            집행위원회
          </h3>
          <p>
            연간 사업계획과 예산을 수립하고, 회칙/규정 개정 절차를 관리합니다. 모든 안건과 지출은 분기 총회에서 공개합니다.
          </p>
        </li>
        <li
          class="about-section__card"
        >
          <h3
            class="about-section__card-title"
          >
            기획분과
          </h3>
          <p>
            산업별 네트워킹, ESG 프로젝트, 장학사업 등 신규 프로그램을 설계합니다. 2025년에는 금융·테크·공공 분야 협업을 우선 추진합니다.
          </p>
        </li>
        <li
          class="about-section__card"
        >
          <h3
            class="about-section__card-title"
          >
            회원분과
          </h3>
          <p>
            회원 데이터베이스 정비와 입회/재등록 지원, 복지 프로그램 운영을 담당합니다. 개인정보 보호와 접근성 가이드를 준수해 서비스 품질을 유지합니다.
          </p>
        </li>
        <li
          class="about-section__card"
        >
          <h3
            class="about-section__card-title"
          >
            소통분과
          </h3>
          <p>
            뉴스레터·SNS·홈페이지 업데이트를 통합 관리하고, 문의 대응 SLA(48시간)를 모니터링합니다. 내·외부 커뮤니케이션 일정을 조율합니다.
          </p>
        </li>
      </ul>
    </section>
    <section
      aria-labelledby="org-contact"
      class="about-section about-section--highlight"
    >
      <div
        class="about-section__content"
      >
        <h2
          class="about-section__heading"
          id="org-contact"
        >
          연락 및 협업 채널
        </h2>
        <p>
          공식 창구를 통해 모든 문의를 받고 있습니다. 회비 관련 문의, 협업 제안, 행사 대관 등은 사무국에서 담당자를 지정하여 응대합니다.
        </p>
      </div>
      <dl
        aria-label="사무국 연락처 정보"
        class="about-contact-list"
      >
        <div
          class="about-contact-list__item"
        >
          <dt
            class="about-contact-list__term"
          >
            사무국 대표 메일
          </dt>
          <dd
            class="about-contact-list__description"
          >
            office@sogang-econ-alumni.kr
          </dd>
        </div>
        <div
          class="about-contact-list__item"
        >
          <dt
            class="about-contact-list__term"
          >
            대표 전화
          </dt>
          <dd
            class="about-contact-list__description"
          >
            02-715-1234 (ARS 2번: 행사/협업)
          </dd>
        </div>
        <div
          class="about-contact-list__item"
        >
          <dt
            class="about-contact-list__term"
          >
            운영시간
          </dt>
          <dd
            class="about-contact-list__description"
          >
            평일 10:00 - 17:00 (점심 12:00 - 13:00)
          </dd>
        </div>
        <div
          class="about-contact-list__item"
        >
          <dt
            class="about-contact-list__term"
          >
            주소
          </dt>
          <dd
            class="about-contact-list__description"
          >
            서울 마포구 백범로 35, 서강대학교 경제대학원 3층 총동문회 사무국
          </dd>
        </div>
      </dl>
    </section>
  </div>
</DocumentFragment>
`;
